### PR TITLE
issue: 844097 vma_get_socket_rings_fds() crash when binding on non-MLX cards

### PR DESCRIPTION
**Pull request by serglelv**
Checking for NULL pointer was added
@igor-ivanov @OphirMunk @rosenbaumalex

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/327/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/345/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/347/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/365/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk @rosenbaumalex it is ready to be merged

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/373/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/374/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/391/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/392/ for details (Mellanox internal link).

File: src/vma/sock/sock-redirect.cpp [hunk 1]
```diff
@@ -380,6 +380,7 @@ int vma_free_vma_packets(struct vma_packet_desc_t *packets, int num)
 			desc = (mem_buf_desc_t*)packets[i].buff_lst;
 			p_socket_object = (socket_fd_api*)desc->path.rx.context;
 			ring* rng = (ring*)desc->p_desc_owner;
+			if (p_socket_object)
```
Comment (igor-ivanov) [inline src/vma/sock/sock-redirect.cpp]: @serglelv please consider:
<pre>
if (p_socket_object) {
   if (rng) {
   }
}
</pre>

File: src/vma/sock/sock-redirect.cpp [hunk 2]
```diff
@@ -475,11 +476,13 @@ int vma_get_socket_rings_fds(int fd, int *ring_fds, int ring_fds_sz)
 	if (ring_fds_sz > 0) {
 		p_socket_object = fd_collection_get_sockfd(fd);
 		if(p_socket_object) {
+			if(p_socket_object->check_rings_fds()) {
 			p_rings_fds = p_socket_object->get_rings_fds();
```
Comment (igor-ivanov) [inline src/vma/sock/sock-redirect.cpp]: please correct indent

File: src/vma/sock/sock-redirect.cpp [hunk 3]
```diff
@@ -474,10 +475,13 @@ int vma_get_socket_rings_fds(int fd, int *ring_fds, int ring_fds_sz)
 
 	if (ring_fds_sz > 0) {
 		p_socket_object = fd_collection_get_sockfd(fd);
-
-		p_rings_fds = p_socket_object->get_rings_fds();
-		*ring_fds = p_rings_fds[0];
-		rings_num = 1;
+		if(p_socket_object) {
+			if(p_socket_object->check_rings_fds()) {
+			p_rings_fds = p_socket_object->get_rings_fds();
```
Comment (igor-ivanov) [inline src/vma/sock/sock-redirect.cpp]: please correct indent

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/377/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/395/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/381/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/399/ for details (Mellanox internal link).

File: src/vma/sock/sock-redirect.cpp (hunk 1 above)
Comment (serglelv) [inline src/vma/sock/sock-redirect.cpp]: This check is unclear for me and assert for rng and desc have been added

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/382/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/400/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Will be merged with the following title:
issue: 844097 Checking for NULL was updated and segfault fixed

**Comment by OphirMunk** [comment]
Merged https://github.com/Mellanox/libvma/pull/168

**Pull request closed without merging**